lib/_http_agent.js

```diff
@@ -115,6 +115,9 @@ Agent.prototype.addRequest = function(req, options) {
     };
   }
 
+  options = util._extend({}, options);
```

Comment:
@benjamingr we usually just update things like that when the code needs to be updated for some other reason.

@saperal ok, feel free to leave this as is.